addons/mail/static/src/messaging/entity/notification_group_manager/notification_group_manager.js

```diff
@@ -0,0 +1,73 @@
+odoo.define('mail.messaging.entity.NotificationGroupManager', function (require) {
+'use strict';
+
+const { registerNewEntity } = require('mail.messaging.entityCore');
+const { one2many } = require('mail.messaging.EntityField');
+
+function NotificationGroupManagerFactory({ Entity }) {
+
+    class NotificationGroupManager extends Entity {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        computeGroups() {
+            if (!this.env.messaging.currentPartner) {
+                this.update({ groups: [['unlink-all']] });
+                return;
+            }
+            const groups = [];
+            this.env.messaging.currentPartner.failureNotifications.forEach(notification => {
+                const thread = notification.message.originThread;
+                // Notifications are grouped by model and notification_type.
+                // Except for channel where they are also grouped by id because
+                // we want to open the actual channel in discuss or chat window
+                // and not its kanban/list/form view.
+                const channelId = thread.model === 'mail.channel' ? thread.id : null;
+                const id = `${thread.model}/${channelId}/${notification.notification_type}`;
+                const group = this.env.entities.NotificationGroup.insert({
+                    id,
```

Comment:
instead of relying on default unicity of entity from `id` and shaping `id` in a way to be uniquely identifiable, I would override `_findFunctionFromData` and make combination of thread model / id / notification type as a way to uniquely identify a notification group. Could also override `_createInstanceLocalId` for easy-debugging with meaningful localId

Reply:
Good idea, this was basically old code from before fields when `id` was the only possible way.

Reply:
After investigation, this is too hard to do now, at least until we have related dependencies.

The main points:
- group manager should compute everything
- every notification created in the system should be linked to that manager
- it should filter the notifs based on current partner being author of related notification message
- it should manage its list of groups internally based on the filtered notifs above and their fields (at least `notification.message.originThread`: `id` and `model`, + `notification_type`)
- later thread and odoobot request should be introduced into that mix somehow...